src/transform/src/analysis/equivalences.rs

```diff
@@ -630,6 +630,8 @@ impl EquivalenceClasses {
                 expr.permute(permutation);
             }
         }
+        self.remap.clear();
```

Comment:
`minimize()` uses `self.remap` before it calls `refresh()`. You are right about the comment; and we should figure out the right way to communicate "`remap` is not valid". Perhaps this method, which 100% invalidates `remap`, should take `self` and produce a `Self`?

Reply:
This change, fwiw, is probably not load bearing. It's a potential glitch I noticed while debugging the apparent issue the code fixes. I'm not sure we are feeling that pain, but .. if you don't clear `self.remap` then we'll use a potentially wildly incorrect map (e.g. unpermuted) to assess the inbound nullability statements.

Perhaps morally `permute` should destroy `self` to prevent the further use of `self.remap`, which is 100% wrong after calls to `permute` and `project`.

Reply:
Umm, so do I understand correctly that `remap` can be just a little wrong, in which case `minimize` is ok to use it before `refresh`, or very wrong, in which case `minimize` might do an incorrect thing if it uses it?

Reply:
The intended use is that you add to `classes`, and `remap` always reflects a valid historical equivalence class structure if this is all you do. This stops being true as soon as start generally mutating classes, dropping or rewriting expressions, which both `permute` and `project` do (and tbh, a few other moments, by independently reducing elements in/around join).

The current `minimize` behavior would be only "lightly wrong" before `refresh()`, in that it might fail to incorporate nullability statements, and shouldn't misoptimize anything as far as I can reason. But that seems very fragile, and a better thing to do when you change the base facts (as in these methods) is to blow away the cached summary of them.

Reply:
How about I remove the bugfix from this PR, to make it less controversial, and we can discuss independent of this merging?

Reply:
Well, the added lines seem not wrong, so it's fine to have them in the PR. I was just trying to see whether we can make the code a bit more clear. But it's fine to do that as a follow-up PR.

Reply:
Happy to do it either way. I'll add some docs about `remap.clear()` being important if you invalidate the data by mutating the classes. My attempts to make the API safer by only allowing an `append_equivs` method has so far failed, but may pick that up again.